### PR TITLE
Bump for PureScript 0.9.2

### PR DESCRIPTION
**Pull request by garyb**

### CODE REVIEW

**Comment by garyb** [comment]
Ah, it's not published on Hackage yet.

**Comment by paf31** [comment]
Fixed, sorry about that.

**Comment by garyb** [comment]
Thanks! Is that you who restarted the builds too?

**Comment by paf31** [comment]
Yes 😄

**Comment by shinnn** [comment]
I just checked out this branch, ran `npm test` and failed with the same error as Travis CI raises.

```
# Build script
Error: cabal sandbox init --sandbox="/Users/Shinnosuke/github/node-purescript-bin/tmp/.cabal-sandbox" && cabal update && cabal install --bindir /Users/Shinnosuke/github/node-purescript-bin/tmp --jobs=8
Command failed: cabal install --bindir /Users/Shinnosuke/github/node-purescript-bin/tmp --jobs=8
cabal: Error: some packages failed to install:
purescript-0.9.2 failed during the building phase. The exception was:
ExitFailure 1

    at ChildProcess.exithandler (child_process.js:202:12)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:191:7)
    at maybeClose (internal/child_process.js:852:16)
    at Socket.<anonymous> (internal/child_process.js:323:11)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:188:7)
    at Pipe._handle.close [as _onclose] (net.js:492:12)not ok 15 Error: cabal sandbox init --sandbox="/Users/Shinnosuke/github/node-purescript-bin/tmp/.cabal-sandbox" && cabal update && cabal install --bindir /Users/Shinnosuke/github/node-purescript-bin/tmp --jobs=8 Command failed: cabal install --bindir /Users/Shinnosuke/github/node-purescript-bin/tmp --jobs=8 cabal: Error: some packages failed to install: purescript-0.9.2 failed during the building phase. The exception was: ExitFailure 1 
```

**Comment by garyb** [comment]
@paf31 any idea what the problem might be here? I guess "cabal".

Maybe we should fall back to a `stack` based build rather than `cabal`, since that's what we encourage people to use if they're building themselves anyway?

**Comment by shinnn** [comment]
> Maybe we should fall back to a `stack` based build rather than `cabal`

👍 Sounds good to me. We should change these lines for switching to `stack`:

*  https://github.com/purescript-contrib/node-purescript-bin/blob/f2bf2b3ce88366d27e13e35639ca018057b66638/lib/install.js#L30-L32
* https://github.com/purescript-contrib/node-purescript-bin/blob/83334915108016833b7be963151d46017e57ded6/test.js#L55-L57

Roughly speaking, is `cabal install` considered obsolete in Haskell ecosystem?

**Comment by garyb** [comment]
> Roughly speaking, is `cabal install` considered obsolete in Haskell ecosystem?

I think it's getting that way, yeah. `stack` completely eliminates "cabal hell" and even deals with installing the appropriate compiler version if you don't already have it. I certainly don't think it's an unreasonable prerequesite to state, it seems to be pretty common from what I've seen. `stack` can even be installed through `cabal` too :smile:.

**Comment by paf31** [comment]
I think Stack is probably the solution here, as you say.

**Comment by garyb** [comment]
Hmm, well that kinda worked. It seems to have timeout issues, but [can pass](https://travis-ci.org/purescript-contrib/node-purescript-bin/jobs/146081844). I'll try extending the `travis_wait` time.

File: lib/install.js [hunk 1]
```diff
@@ -27,9 +27,8 @@ eachSeries(Object.keys(paths), (key, next) => {
 
       binBuild()
         .src(SOURCE_URL)
-        .cmd(`cabal sandbox init --sandbox="${path.join(bin.dest(), '.cabal-sandbox')}"`)
-        .cmd('cabal update')
-        .cmd(`cabal install --bindir ${bin.dest()} --jobs=${os.cpus().length}`)
+        .cmd('stack setup')
+        .cmd(`stack install --local-bin-path ${bin.dest()} --jobs=${os.cpus().length}`)
```
Comment (shinnn) [inline lib/install.js]: @garyb, would you mind checking whether `--jobs` actually makes build faster in `stack install`?

Comment (garyb) [inline lib/install.js]: I'm not sure if it makes it faster, but it does still enable parallel builds/tests - what do you think, just try taking it out and see if it's better?

**Comment by garyb** [comment]
Looks like the cache worked though! 5/6 mins for the rebuilds here: https://travis-ci.org/purescript-contrib/node-purescript-bin/builds/146087043

File: lib/install.js (hunk 1 above)
Comment (shinnn) [inline lib/install.js]: https://github.com/commercialhaskell/stack/blob/bcb8b4c7929a8f6a39e030fa6ec7ced2cbbbbbd7/src/Stack/Types/Config.hs#L287

> defaults to number of capabilities

It seems that we don't need to specify it explicitly. Can you remove `--jobs` flag?

File: test.js [hunk 2]
```diff
@@ -49,12 +48,12 @@ test('Build script', t => {
   const tmpDir = path.join(__dirname, 'tmp');
 
   rimraf.sync(tmpDir);
+  fs.mkdirSync(tmpDir);
```
Comment (shinnn) [inline test.js]: I just ran this test locally and understood why this line is needed. Hope `stack` handle it gracefully...

**Comment by garyb** [comment]
Looks like it's all good now. Are you happy with things the way they are now @shinnn? (Once Travis passes again... I squashed the commits fiddling with the stack & build options).

**Comment by shinnn** [comment]
LGTM. Great work, @garyb!

**Comment by garyb** [comment]
Thanks!

**Pull request merged**